pkg/sql/sqlbase/table.go

```diff
@@ -1669,6 +1733,10 @@ func encodeArrayElement(b []byte, d parser.Datum) ([]byte, error) {
 	case *parser.DOid:
 		return encoding.EncodeUntaggedIntValue(b, int64(t.DInt)), nil
 	}
+	if d == parser.DNull {
```

Comment:
leave this condition out. The catch-all error below will reveal nulls just as well.